astropy/visualization/wcsaxes/core.py

```diff
@@ -197,17 +198,25 @@ def imshow(self, X, *args, **kwargs):
 
         # To check whether the image is a PIL image we can check if the data
         # has a 'getpixel' attribute - this is what Matplotlib's AxesImage does
-
         try:
             from PIL.Image import Image
-            from PIL.Image.Transform import FLIP_TOP_BOTTOM
```

Comment:
I believe, what @pllim is suggesting is replace this with:
```python
            if minversion("PIL", "9.1"):
                from PIL.Image.Transform import FLIP_TOP_BOTTOM
            else:
                from PIL.Image import FLIP_TOP_BOTTOM
```

Reply:
Hmm this package should have used `HAS_PIL` from https://github.com/astropy/astropy/blob/main/astropy/utils/compat/optional_deps.py but that is out of scope here.

Reply:
Agreed

Reply:
I've updated this PR to use `HAS_PIL`.  I'm not sure why that exists because it's not an optional dependency.

Reply:
As far as I am aware Pillow, is not a required dependency for astropy. Indeed, I don't see it anywhere in `setup.cfg`:
https://github.com/astropy/astropy/blob/f5f39c0fff49347886f6d4e7670d7e67e37fd081/setup.cfg#L30-L103
nor is it installed via a `pip install astropy` in a clean environment.

Reply:
I think `visualization` pulls it in or something. I dunno. If only we can remove that subpackage from core... 😏